BuildResidentialHPXML/resources/geometry.rb

```diff
@@ -63,43 +63,30 @@ def self.create_single_family_detached(runner, model,
     foundation_height = UnitConversions.convert(foundation_height, 'ft', 'm')
     rim_joist_height = UnitConversions.convert(rim_joist_height, 'ft', 'm')
 
-    garage_area = garage_width * garage_depth
-    has_garage = false
-    if garage_area > 0
-      has_garage = true
-    end
+    length, width = calculate_footprint_length_width(garage_width, garage_depth, garage_protrusion, foundation_type, attic_type, num_floors, cfa, aspect_ratio)
 
-    # calculate the footprint of the building
-    garage_area_inside_footprint = 0
-    if has_garage
-      garage_area_inside_footprint = garage_area * (1.0 - garage_protrusion)
+    # Error checking: adjust garage width if too wide relative the living space dimensions
+    # e.g., this can happen if garage protrudes too much for a small dwelling unit
+    garage_width_adjusted = false
+    while ((garage_width >= length) && (garage_depth > 0))
```

Comment:
Is it possible to use math to calculate the max garage dimension size and not have to iterate?

Reply:
Not worth the effort.